tribe-ext-disallow-past-starting-dates.php

```diff
@@ -0,0 +1,517 @@
+ * Author URI:        http://m.tri.be/1971
+ * License:           GPL version 3 or any later version
+ * License URI:       https://www.gnu.org/licenses/gpl-3.0.html
+ * Text Domain:       tribe-ext-custom-datepicker-start-date
+ *
+ *     This plugin is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     any later version.
+ *
+ *     This plugin is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *     GNU General Public License for more details.
+ */
+
+/**
+ * Known issues:
+ * 1)
+ * If the user's local time zone and the site's time zone are on different
+ * calendar days (such as UTC+0 and UTC+1 and UTC is at 23:00:00), the
+ * datepicker will display the user's today as selectable but it won't be
+ * selectable. This is correct because the site owner does not want the user to
+ * select their own today because it's the site's yesterday.
+ *
+ * This odd UX is due to the jQuery UI Datepicker library not supporting
+ * time zones and the timepicker being a separate input.
+ *
+ * 2)
+ * All "protection" against setting an invalid start date is in JavaScript--none
+ * in PHP.
+ */
+
+// Do not load unless Tribe Common is fully loaded and our class does not yet exist.
+if ( class_exists( 'Tribe__Extension' ) && ! class_exists( 'Tribe__Extension__Custom_Datepicker_Start_Date' ) ) {
+	/**
+	 * Extension main class, class begins loading on init() function.
+	 */
+	class Tribe__Extension__Custom_Datepicker_Start_Date extends Tribe__Extension {
+		/**
+		 * The script's handle.
+		 */
+		public $handle = 'tribe-ext-custom-datepicker-start-date';
+
+		/**
+		 * The array of keys/values to send to the script.
+		 *
+		 * @see wp_localize_script()
+		 *
+		 * @var array
+		 */
+		private $script_vars = array();
+
+		/**
+		 * The existing start date's midnight timestamp.
+		 *
+		 * This will remain NULL if this is a new (non-existing) event or if its
+		 * start date is TODAY.
+		 *
+		 * @see Tribe__Extension__Custom_Datepicker_Start_Date::get_min_date()
+		 *
+		 * @var null|int
+		 */
+		private $timestamp_existing_start_date = null;
+
+		/**
+		 * The timestamp of whichever is greater between Today and an existing
+		 * event's start time, if applicable.
+		 *
+		 * This will remain NULL if this is a new (non-existing) event or if its
+		 * start date is TODAY.
+		 *
+		 * @see Tribe__Extension__Custom_Datepicker_Start_Date::get_min_date()
+		 *
+		 * @var null|int
+		 */
+		private $timestamp_max_today_or_existing = null;
+
+		/**
+		 * The script's handle with underscores instead of hyphens.
+		 *
+		 * Used for building filter and action hook names and CSS class name.
+		 *
+		 * @return string
+		 */
+		public function get_handle_underscores() {
```

Comment:
How about setting this during "construction" and storing in a private field. Then, when we need this, we can save ourselves a method call.

Reply:
removed the method; didn't replace it